scene/main/window.cpp

```diff
@@ -1383,6 +1385,32 @@ void Window::_window_input(const Ref<InputEvent> &p_ev) {
 		emit_signal(SceneStringNames::get_singleton()->window_input, p_ev);
 	}
 
+	if (!Engine::get_singleton()->is_editor_hint() && p_ev->is_action_pressed("ui_take_screenshot")) {
+		// The editor already has its own screenshot taking functionality.
+		const Ref<ViewportTexture> texture = get_texture();
+		ERR_FAIL_COND_MSG(texture.is_null(), "Cannot get a viewport texture from the window for taking a screenshot.");
+		const Ref<Image> img = texture->get_image();
+		ERR_FAIL_COND_MSG(img.is_null(), "Cannot get an image from a viewport texture of the window for taking a screenshot.");
+
+		Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+		String project_name = GLOBAL_GET("application/config/name");
+		if (project_name.is_empty()) {
+			project_name = "[unnamed project]";
+		}
+		// Use a subfolder with the project name in the user's Pictures folder for easier discovery of the screenshots taken.
+		const String path = OS::get_singleton()->get_system_dir(OS::SYSTEM_DIR_PICTURES).path_join(project_name).path_join("screenshot_" + Time::get_singleton()->get_datetime_string_from_system().replace(":", "") + ".png");
+		da->make_dir_recursive(path.get_base_dir());
+		const Error error = img->save_png(path);
+
+		ERR_FAIL_COND_MSG(error != OK, vformat("Cannot save screenshot to: %s", path));
+		// Using formatting around the printed path allows using Ctrl + Click in VS Code's integrated terminal to open the screenshot:
+		// <https://github.com/microsoft/vscode/issues/97941>
+		print_line_rich(vformat("Screenshot saved to: [u]%s[/u]", path));
+
+		// Prevent embedded subwindows from also taking a screenshot.
+		set_input_as_handled();
```

Comment:
My previous comment was apparently ambiguous, because you still propagate the event via `push_input` even if it is set as handled here (which could cause the InputEvent to be used a second time). Sorry for that!
In order to prohibit that, please `return;` instead of `set_input_as_handled();` or introduce a `!is_input_handled()` check before `push_input`.